nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/proxy/P2ProxyFacetImpl.java

```diff
@@ -163,23 +167,48 @@ private Content removeMirrorUrlFromArtifactsAndSaveMetadataAsAsset(final String
 
     String assetPath = path;
 
-    if (assetKind.equals(AssetKind.ARTIFACT_XML)) {
-      try (TempBlob newMetadataContent = xmlRewriter.removeMirrorUrlFromArtifactsXml(metadataContent, getRepository(), "xml")) {
-        return saveMetadataAsAsset(assetPath, newMetadataContent, payload, assetKind);
-      }
-    }
-    else if (assetKind.equals(AssetKind.ARTIFACT_JAR)) {
-      try (TempBlob newMetadataContent = xmlRewriter.removeMirrorUrlFromArtifactsXml(metadataContent, getRepository(), "jar")) {
-        return saveMetadataAsAsset(assetPath, newMetadataContent, payload, assetKind);
-      }
-    }
-    else if (assetKind.equals(AssetKind.ARTIFACT_XML_XZ)) {
-      try (TempBlob newMetadataContent = xmlRewriter.removeMirrorUrlFromArtifactsXml(metadataContent, getRepository(), "xml.xz")) {
-        return saveMetadataAsAsset(assetPath, newMetadataContent, payload, assetKind);
-      }
-    }
-    else {
-      return saveMetadataAsAsset(assetPath, metadataContent, payload, assetKind);
+    switch (assetKind) {
+      case ARTIFACT_XML:
+        try (TempBlob newMetadataContent = xmlRewriter
```

Comment:
Question - Is it possible to move outside try block opening? And make one outgoing point (one return)?

Reply:
Done